include/libtorrent/netlink.hpp

```diff
@@ -63,7 +63,7 @@ namespace libtorrent
 			std::memset(&m_sockaddr, 0, sizeof(sockaddr_nl));
 			m_sockaddr.nl_family = AF_NETLINK;
 			m_sockaddr.nl_groups = group;
-			m_sockaddr.nl_pid = pid;
+			m_sockaddr.nl_pid = std::uint32_t(pid);
```

Comment:
also, as far as I can tell, there's no place where the "port ID" is actually specified. how about you just make the parameter a ``uint32_t`` instead? That seems to be the most correct solution.